packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -198,14 +198,25 @@ class DependencyExtractionWebpackPlugin {
 				}
 			}
 
+			// Go through the assets and hash the sources. We can't just use
+			// `entrypointChunk.contentHash` because that's not updated when
+			// assets are minified. Sigh.
+			// @todo Use `asset.info.contenthash` if we can make sure it's reliably set.
+			const hash = createHash( 'sha512' );
+			for ( const filename of entrypoint.getFiles().sort() ) {
+				const asset = compilation.getAsset( filename );
+				hash.update( `${ filename }: ` );
```

Comment:
Does it mean that the the hash generated in the asset file depends now only on the list of sorted file names imported in the entry?

Reply:
Calling `hash.update()` with the contents of the asset.

Reply:
Now that I think about it I can see where your confusion may have come from, it's kind of backwards that `asset.source.updateHash( hash )` updates the passed `hash` rather than the `asset.source`, and https://www.npmjs.com/package/webpack-sources#updateHash doesn't seem to come up in the Google searches I tried (I only found it just now when grepping the code for docs).

Reply:
Thank you. Now it all makes sense. I tested it and it works perfectly fine.

I tested with imported CSS files and changes to the content impact the hash generated for the JS file, but it's similar to the current behavior when using `entrypointChunk.hash`. I guess that's fine.